collections/soldiers.js

```diff
@@ -0,0 +1,47 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+const configs = require('../config');
+const utils = require("../utility/utility");
+
+// Connection URL
+const url = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+// Use connect method to connect to the server
+function newSoldier(collection, data, errorHandler) {
+  if (data["id"] && !(utils.isMatchesRegex(/[0-9]{7}/, data["id"]))) {
+    errorHandler.notValidData("The soldier json that was sent didnt had a legal id");
+  } else {
+    let fields = ["id", "name", "rank", "limitations"];
+
+    for (let i = 0; i < fields.length; i++) {
+      if (!data[fields[i]]) {
+        errorHandler.notValidData("The soldier json that was sent is missing the field " + fields[i]);
+        return;
+      }
+    }
+
+    if (!Array.isArray(data["limitations"])) {
+      errorHandler.notValidData("The soldier json that was sent didnt had a legal limitations field (it was not an Array)");
```

Comment:
"limitations must be an array."

in your error messages try to think what the user knows and what he doesn;t. he knows he sent a soldier json and that there was a problem. so telling him that is redundant. the problem itself "limitations must be an array." is all he needs